src/python/pants/engine/environment.py

```diff
@@ -17,12 +17,21 @@
 name_value_re = re.compile(r"([A-Za-z_]\w*)=(.*)")
 shorthand_re = re.compile(r"([A-Za-z_]\w*)")
 
+"""
+Accesses to `os.environ` cannot be accurately tracked, so @rules that need access to the
+environment should use APIs from this module instead.
+
+Wherever possible, the `Environment` type should be consumed rather than the
+`CompleteEnvironment`, as it represents a filtered/relevant subset of the environment, rather
```

Comment:
Should we rename `CompleteEnvironment` to `CompleteEnviroment__Slow` or something?

Reply:
Maybe. Or move it out of this module and into the `internals`? It is widely used (within internals), but I looked for other usage in `@rules`, and this was the only case.

Reply:
Yes, good idea! And I recommend renaming the import in this file to be private, like `import CompleteEnvironment as _CompleteEnvironment`, given Python's re-export semantics.

Or, just keep it here and call it `_CompleteEnvironment`?